frontend/lib/src/components/core/Block/Block.tsx

```diff
@@ -72,11 +72,14 @@ const ChildRenderer = (props: BlockPropsWithoutWidth): ReactElement => {
 
   // Capture all the element ids to avoid rendering the same element twice
   const elementKeySet = new Set<string>()
+  let clearableCount = 0
+  let blockCount = 0
 
   return (
     <>
       {props.node.children &&
         props.node.children.map((node: AppNode, index: number): ReactNode => {
+          const indexOffset = clearableCount + blockCount
```

Comment:
nit: might be good to add a comment here to explain why the offset. Also, something like `transientElementCount` might be a better name for `clearableCount`